ranger-http/src/main/java/io/appform/ranger/http/servicefinder/HttpCommunicator.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2024 Authors, Flipkart Internet Pvt. Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.appform.ranger.http.servicefinder;
+
+import io.appform.ranger.core.model.Service;
+import io.appform.ranger.core.model.ServiceNode;
+import io.appform.ranger.http.serde.HTTPResponseDataDeserializer;
+import okhttp3.OkHttpClient;
+
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Interface for communicator to upstream
+ */
+public interface HttpCommunicator<T> extends AutoCloseable {
```

Comment:
To what I'm seeing, it is more of a NodeDataSource interface - functionally - than an HttpCommunicator. With both getServices and listNodes being part of that interface, further can be of types, zk, http, drove or whatever backend in the future. Practically combining both ServiceDataSource and NodeDataSource into a singular functional interface - which is understandable. And an httpCommunicator being a member of such an httpNodeDataSource is gettable. 

In this case, I don't see ServiceDataSource done away with. Is that the future idea? Since where you got the service list is where you will typically find your nodeList?

Reply:
The argument would only be on StaticServiceDataSource. That could be a property in the functional interface then?

Reply:
Either way, it is not counter-intuitive even if the interfaces stay as is, for now!